fairlearn/reductions/_moments/utility_parity.py

```diff
@@ -282,12 +374,30 @@ class TruePositiveRateParity(UtilityParity):
 
     short_name = "TruePositiveRateParity"
 
-    def load_data(self, X, y, **kwargs):
-        """Load the specified data into the object."""
+    def compute_base_event(self, X: pd.DataFrame, y: pd.Series) -> pd.Series:
+        """Compute the basic event for false positive parity.
+
+        This is an array with the string :code:`LABEL=1` where :code:`y` is
+        one.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            The feature array as a :class:`pandas.DataFrame`
+        y : pd.Series
+            The label vector as a :class:`pandas.Series`
+
+        Returns
+        -------
+        pandas.Series
+            The event vector
+        """
+        assert isinstance(X, pd.DataFrame)
```

Comment:
I think that at some point we might have decided not to use `assert` in our libraries and only in tests? @romanlutz @adrinjalali ?

Reply:
[but this is not a biggie... i'm fine leaving as is for now]

Reply:
Not a great way to do error handling... at least there should be a message to be shown if the assert fails. What's the argument against raising exceptions?

Reply:
I used `assert` because this is not an expected error condition, but an internal sanity check. This should never be called unless the arguments have been converted.

Reply:
See note above about what the public API is supposed to be. This particular point is moot, though since I'm about to remove this method.